app/server.cjsx

```diff
@@ -0,0 +1,142 @@
+      'Intl': flags: ['gated']
+      'Intl.~locale.en': flags: ['gated']
+      'Intl.~locale.fi': flags: ['gated']
+      'Intl.~locale.sv': flags: ['gated']
+      'Object.assign': flags: ['gated']
+      'Array.prototype.find': flags: ['gated']
+      'es5': flags: ['gated']
+    minify: true
+    unknown: 'polyfill'
+
+getScripts = ->
+  if config.NODE_ENV == 'development'
+    <script async src="//localhost:9000/js/bundle.js"/>
+  else
+    [
+      <script dangerouslySetInnerHTML={ __html: manifest }/>,
+      <script src={config.APP_PATH + '/' + stats.assetsByChunkName.common[0]}/>,
+      <script src={config.APP_PATH + '/' + stats.assetsByChunkName.leaflet[0]}/>,
+      <script src={config.APP_PATH + '/' + stats.assetsByChunkName.main[0]}/>
+    ]
+
+getContent = (context, renderProps, locale) ->
+  # Ugly way to see if this is a Relay RootComponent
+  # until Relay gets server rendering capabilities
+  if renderProps.components.some(((i) -> i instanceof Object and i.getQuery))
+    return ''
+
+  # TODO: This should be moved to a place to coexist with similar content from client.cjsx
+  ReactDOM.renderToString(
+    <FluxibleComponent context={context.getComponentContext()}>
+      <IntlProvider locale={locale} messages={translations[locale]}>
+        <RoutingContext
+          history={renderProps.history}
+          createElement={React.createElement}
+          location={renderProps.location}
+          routes={renderProps.routes}
+          params={renderProps.params}
+          components={renderProps.components} />
+      </IntlProvider>
+    </FluxibleComponent>
+  )
+
+getHtml = (context, renderProps, locale, polyfills) ->
+  ReactDOM.renderToStaticMarkup <ApplicationHtml
+    css={if config.NODE_ENV == 'development' then false else css}
+    svgSprite={svgSprite}
+    content={getContent(context, renderProps, locale)}
+    polyfill={polyfills}
+    state={'window.state=' + serialize(application.dehydrate(context)) + ';'}
+    livereload={if config.NODE_ENV == 'development' then '//localhost:9000/' else config.ROOT_PATH + '/'}
+    locale={'window.locale="' + locale + '"'}
+    scripts={getScripts()}
+    fonts={fonts}
+    config={'window.config=' + JSON.stringify(config)}
+  />
+
+module.exports = (req, res, next) ->
+  # pass in `req.url` and the router will immediately match
+  locale = req.cookies.lang or req.acceptsLanguages(['fi', 'sv', 'en']) or 'en'
+  context = application.createContext()
+  #required by material-ui
+  global.navigator = userAgent: req.headers['user-agent']
+  location = useBasename(useQueries(createHistory))(basename: config.ROOT_PATH).createLocation(req.url)
+
+  match {routes: context.getComponent(), location: location}
+  , (error, redirectLocation, renderProps) ->
+    if redirectLocation
+      res.redirect 301, redirectLocation.pathname + redirectLocation.search
+    else if error
+      return next(error)
+    else if !renderProps
+      res.status(404).send 'Not found'
```

Comment:
Should we remove:
- https://github.com/HSLdevcom/digitransit-ui/blob/split-vendor-chunks/app/page/404.cjsx